R/calcElecDemandICT.R

```diff
@@ -0,0 +1,207 @@
+#' Calculate Electricity Demand for ICT from Data Center Demands
+#'
+#' This function processes historical and scenario electricity demand data for ICT
+#' (Information and Communication Technology) and data centers. The initial data
+#' comes from exogenous projections using the Digital Transformation Level (DGI)
+#' as the primary driver of demand, with IEA data serving as the historical baseline.
+#'
+#' The function processes multiple demand estimates:
+#' \itemize{
+#'   \item Mean and central estimate
+#'   \item Lower demand boundary (DGI elasticity from historical efficiency dominant phases)
+#'   \item Upper demand boundary (DGI elasticity from historical service dominant phases)
+#' }
+#'
+#' Data is downscaled from R5 to country-level using historical data center counts,
+#' with fixed network shares calculated at the end of the historical period. Historical
+#' data (IEA Base) is separated and harmonized with SSP scenario projections.
+#'
+#' @param endOfHistory upper temporal boundary for historical data
+#'
+#' @returns A magpie object containing ICT electricity demand data calculated from
+#'   data center demands across regions, time periods, and scenarios.
+#'
+#' @author Hagen Tockhorn
+#'
+#' @importFrom madrat readSource toolGetMapping
+#' @importFrom tibble as_tibble
+#' @importFrom dplyr filter mutate group_by reframe ungroup rename
+#'   select across all_of .data %>% cross_join left_join
+#' @importFrom tidyr pivot_wider pivot_longer replace_na
+#' @importFrom purrr map_dfr
+#' @importFrom quitte as.quitte interpolate_missing_periods
+#' @importFrom magclass as.magpie
+#' @importFrom madrat toolCountryFill
+
+calcElecDemandICT <- function(endOfHistory = 2025) {
+
+  # PARAMETER ------------------------------------------------------------------
+
+  # TWh to EJ conversion factor
+  twh2ej <- 3600 * 10^12 / 10^18
+
+
+
+  # READ-IN DATA ---------------------------------------------------------------
+
+  # R12 Resolution (Elec. Demand DC + ICT)
+  dataR12 <- readSource("PRISMA_ICT", subtype = "R12", convert = FALSE) %>%
+    as_tibble() %>%
+    filter(!is.na(.data$value))
+
+  # R5 Resolution (Elec. Demand DC)
+  dataR5 <- readSource("PRISMA_ICT", subtype = "R5", convert = FALSE) %>%
+    as_tibble() %>%
+    filter(!is.na(.data$value))
+
+  # Number of DCs
+  dataDC <- readSource("PRISMA_ICT", subtype = "nDC") %>%
+    as_tibble() %>%
+    filter(!is.na(.data$value))
+
+
+  # IEA R5 region mapping
+  regionmap <- toolGetMapping("regionmappingIEA_R5.csv",
+                              type = "regional",
+                              where = "mredgebuildings")
+
+
+
+  # PROCESS DATA ---------------------------------------------------------------
+
+  # calculate global network share from R12 data
+  networkShare <- dataR12 %>%
+    filter(.data$period == endOfHistory) %>%
```

Comment:
If there is temporal information, why do you assume a constant network/DC ratio?